hibernate-core/src/main/java/org/hibernate/resource/transaction/backend/jdbc/internal/JdbcResourceLocalTransactionCoordinatorImpl.java

```diff
@@ -84,8 +84,7 @@ public void explicitJoin() {
 
 	@Override
 	public boolean isJoined() {
-		return physicalTransactionDelegate != null && physicalTransactionDelegate.getStatus() == TransactionStatus.ACTIVE;
-
+		return physicalTransactionDelegate != null && getTransactionDriverControl().isActive( true );
```

Comment:
I do seem to recall that `isTransactionInProgress()` checked the result from `isJoined()` and it was returning `false` because it explicitly checked only for `ACTIVE`.  I changed that line as shown so that when the status was `MARKED_ROLLBACK`, it would yield `true`.

I'll go back and double check my changes for that use case to be certain, however.